### PR TITLE
davmail: initial commit

### PR DESCRIPTION
**Pull request by wberrier**
* Shows main window at launch instead of non-existing systray for
  gnome (patch, not sure if it can be sent upstream)
* desktop/icon works, passes validation
* allows access to home dir (attempts using --persist with
  individual files failed: ~/.davmail.properties and ~/davmail.log*)
* Use cleanup commands to "relocate" app (build system doesn't support
  any notion of "--prefix")
* Insert svn revision into build script when building from tarball to
  fix latest version checks at runtime
* Add patch to fixup desktop categories (will submit upstream)

### CODE REVIEW

**Comment by wberrier** [comment]
Upstream feature-request requesting davmail as flatpak:

https://sourceforge.net/p/davmail/feature-requests/116/

**Comment by nedrichards** [comment]
bot, build org.davmail.DavMail

**Comment by wberrier** [comment]
Turns out I was incorrectly installing the appdata file.  I'm assuming I can't initiate a build?

bot, build org.davmail.DavMail

**Comment by nedrichards** [comment]
bot, build org.davmail.DavMail

**Comment by wberrier** [comment]
Looks like the build succeeded:

https://flathub.org/builds/#/builders/2/builds/4890

**Comment by wberrier** [comment]
@nedrichards thanks for the review, I populated the releases for the appdata.  I'll work with upstream to see how to integrate appdata.

**Comment by nedrichards** [comment]
bot, build org.davmail.DavMail

**Comment by nedrichards** [comment]
@wberrier I've tested this locally and when I run the app I get a window coming up that says:
> Unable to bind server socket for POP on port 1110: port not allowed or in use by another process
Which I assume is some sort of user focused error (maybe I didn't set it up correctly but I got that out of the box with the default outlook settings)
When pressing help I got:
> Unable to open link The BROWSE action is not supported on the current platform!
Other than that the main issue is the automagic release tag appdata, you should just keep the last two or three. The big unformatted blob is sort of unhelpful for users as well, but I understand your point there. I've pushed a commit that makes the size much more reasonable, albeit a bit hand tuned for now. Other than that, the system seems to work. The configuration issue isn't a blocker assuming that you can assure me that DavMail users understand how it works and I suspect the BROWSER thing is just that you should set an env var to xdg-open or similar (I believe that is correctly intercepted by the OpenURI portal inside a flatpak)

**Comment by wberrier** [comment]
> Unable to bind server socket for POP on port 1110: port not allowed or in use by another process

I've only seen the bind error when another program was conflicting on that port (I actually had two instances of davmail running).  Not sure what may be going on otherwise.  But, I've been successfully using the flatpak davmail on both Ubuntu and Fedora without issue.
 
> Unable to open link The BROWSE action is not supported on the current platform!

I looked at the source and it's using a java library for launching a browser.  I'll pursue a solution for this.  Good catch.
 
> Other than that the main issue is the automagic release tag appdata, you should just keep the last two or three. The big unformatted blob is sort of unhelpful for users as well, but I understand your point there. I've pushed a commit that makes the size much more reasonable, albeit a bit hand tuned for now. Other than that, the system seems to work.

That makes sense, I'm good with that.

**Comment by nedrichards** [comment]
libgnome is certainly a GNOME 2 era library so I don't believe it's in the FDo 1.6 runtime unless you bundle it yourself. I suspect that would wend its way through to something like gnome-open or xdg-open and "just work" but it'd need a bit of a dig.

**Comment by nedrichards** [comment]
> I've only seen the bind error when another program was conflicting on that port (I actually had two instances of davmail running). Not sure what may be going on otherwise. But, I've been successfully using the flatpak davmail on both Ubuntu and Fedora without issue.

OK, lgtm

**Comment by mguessan** [comment]
This help button is definitely not mandatory to use DavMail => you could just remove it.

However you will probably soon have another challenge: make JavaFX based interactive authentication work with flatpak.

I managed to RPM build on various distributions, but had to disable JavaFX on most, see:
https://build.opensuse.org/package/show/home:mguessan:davmail-trunk/davmail-trunk

File: org.davmail.DavMail.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
+{
+  "app-id": "org.davmail.DavMail",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "1.6",
```
Comment (nedrichards) [inline org.davmail.DavMail.json]: This should now be changed to 18.08 (OpenJDK on flathub has been updated to support that)

**Comment by nedrichards** [comment]
@wberrier ah, interesting. Especially since OpenJDK11 is currently in review on Flathub and ideally we'd use that (since it's LTS-ified...)

File: org.davmail.DavMail.json (hunk 1 above)
Comment (wberrier) [reply org.davmail.DavMail.json]: Updated to 18.08

**Comment by wberrier** [comment]
@mguessan are you saying we'd also need to bundle javafx?

As for opening browser urls, I'm trying to understand:

* how awt decides how to open a url in a browser on linux
* what a flatpak app needs to do to open up a url in the host system outside the sandbox (dbus based? shelling out to xdg-open?)

Also, another option could be adding another browser backend in davmail that works with flatpak (although I'm hoping that one of the existing backends will work with flatpak, and want to avoid adding a backend that only works with flatpak).

**Comment by nedrichards** [comment]
bot, build org.davmail.DavMail

**Comment by wberrier** [comment]
I added quick and dirty support for xdg-open (after trying to figure out which gtk/gnome libs would be necessary for awt to work).

It works for this case, but may need to be cleaned up before ready to send upstream.

**Comment by nedrichards** [comment]
bot, build org.davmail.DavMail

**Comment by nedrichards** [comment]
bot, build org.davmail.DavMail

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.davmail.DavMail

**Comment by wberrier** [comment]
@nedrichards Now that the davmail repository has been created, what's the next step for getting the app on flathub.org?

**Comment by barthalion** [comment]
Started new build here: https://flathub.org/builds/#/builders/2/builds/5452

Apparently buildbot never received payload via webhook to start it.

**Pull request closed without merging**